lib/getUserConstructor.js

```diff
@@ -0,0 +1,268 @@
+      }
+    }
+
+    /**
+     * @returns {Promise<null|KuzzleUser>}
+     */
+    async getKuzzleUser() {
+      if (!this.kuid) {
+        return null;
+      }
+
+      if (!this[_kuzzleUser]) {
+        try {
+          this[_kuzzleUser] = await plugin.context.accessors.sdk.security.getUser(this.kuid);
+        }
+        catch (e) {
+          if (e.id === 'services.storage.not_found') {
+            this[_kuzzleUser] = null;
+          }
+          else {
+            throw e;
+          }
+        }
+      }
+
+      return this[_kuzzleUser];
+    }
+
+    /**
+     * @returns {Promise<Array<KuzzleUserProfile>>}
+     */
+    async getKuzzleUserProfiles() {
+      if (!this.kuid) {
+        return [];
+      }
+
+      if (!this[_kuzzleUserProfiles]) {
+        const kuzzleUser = await this.getKuzzleUser();
+        if (kuzzleUser === null) {
+          return [];
+        }
+
+        this[_kuzzleUserProfiles] = await plugin.context.accessors.sdk.security.mGetProfiles(
+          kuzzleUser.content.profileIds
+        );
+      }
+      return this[_kuzzleUserProfiles];
+    }
+
+    /**
+     * @returns {Promise<Number>}
+     */
+    async getPasswordRetention() {
+      let retention = 0;
+
+      for (const policy of await this.getPolicies()) {
+        if (policy.forbidReusedPasswordCount) {
+          retention = Math.max(retention, policy.forbidReusedPasswordCount);
+        }
+      }
+
+      return retention;
+    }
+
+    /**
+     * @returns {Promise<Array<PasswordPolicy>>}
+     */
+    async getPolicies() {
+      if (!this.kuid) {
+        return [];
+      }
+
+      const policies = [];
+
+      for (const policy of plugin.config.passwordPolicies) {
+        if (policy.appliesTo === '*') {
+          policies.push(policy);
+        }
+
+        if (policy.appliesTo.users && policy.appliesTo.users.includes(this.kuid)) {
+          policies.push(policy);
+        }
+
+        const kuzzleUser = await this.getKuzzleUser();
+        if (kuzzleUser === null) {
+          continue;
+        }
+
+        if (policy.appliesTo.profiles) {
+          for (const kuzzleUserProfile of await this.getKuzzleUserProfiles()) {
+            if (policy.appliesTo.profiles.includes(kuzzleUserProfile._id)) {
+              policies.push(policy);
+            }
+          }
+        }
+
+        if (policy.appliesTo.roles) {
+          for (const profile of await this.getKuzzleUserProfiles()) {
+            for (const {roleId} of profile.policies) {
+              if (policy.appliesTo.roles.includes(roleId)) {
+                policies.push(policy);
+              }
+            }
+          }
+        }
+      }
+
+      return policies;
+    }
+
+    /**
+     * @returns {Promise<boolean>}
+     */
+    async isPasswordExpired() {
+      for (const policy of await this.getPolicies()) {
+        if (policy.expiresAfter
+          && Date.now() > (this._kuzzle_info.updatedAt || this._kuzzle_info.createdAt) + ms(policy.expiresAfter)
```

Comment:
`expiresAfter` doesn't seem to be defaulted when loading the policies from the plugin configuration: if policy.expiresAfter is null or undefined, `ms` throws an error